src/button/component.jsx

```diff
@@ -614,6 +614,46 @@ export let Button : Component<ButtonOptions> = create({
             }
         },
 
+        onShippingChange: {
+            type:     'function',
+            required: false,
+            timeout:  __TEST__ ? 500 : 10 * 1000,
+            noop:     true,
+
+            decorate(original) : Function {
+                return function decorateOnShippingChange(data, actions) : ZalgoPromise<void> {
+
+                    info('button_shipping_change');
+
+                    track({
+                        [ FPTI.KEY.STATE ]:              FPTI.STATE.CHECKOUT,
+                        [ FPTI.KEY.TRANSITION ]:         FPTI.TRANSITION.CHECKOUT_SHIPPING_CHANGE,
+                        [ FPTI.KEY.BUTTON_SESSION_UID ]: this.props.buttonSessionID
+                    });
+
+                    flushLogs();
+                    let timeout = __TEST__ ? 500 : 10 * 1000;
+
+                    let reject = (...args) => {
+                        if (actions.reject && typeof actions.reject === 'function') {
+                            return ZalgoPromise.try(() => {
+                                return actions.reject.call(null, ...args);
+                            });
+                        }
+                    };
+
+                    return ZalgoPromise.try(() => {
+                        return original.call(this, data, { ...actions, reject });
+                    }).timeout(timeout, new Error(`Timed out waiting ${ timeout }ms for payment`)).catch(err => {
+                        if (this.props.onError) {
```

Comment:
I'd say let's not call the `onError` prop here, since we're handling the error in hermes and showing an error page. Typically for those cases we don't propagate the failure back up to the merchant.